vita3k/touch/src/touch.cpp

```diff
@@ -189,6 +211,57 @@ int handle_touch_event(SDL_TouchFingerEvent &finger) {
     return 0;
 }
 
+int handle_touchpad_event(SDL_ControllerTouchpadEvent &touchpad) {
+    switch (touchpad.type) {
+    case SDL_CONTROLLERTOUCHPADDOWN:
+        if (finger_count >= 2) // best we can do is clean everything
+            finger_count = 0;
+
+        touchpad_buffer[finger_count] = touchpad;
+        touchpad_buffer[finger_count].which = next_touch_id;
+        next_touch_id = (next_touch_id + 1) % 128;
+        finger_count++;
+        break;
+    case SDL_CONTROLLERTOUCHPADUP:
+        for (uint32_t i = 0; i < finger_count; i++) {
+            if (touchpad.finger == touchpad_buffer[i].finger) {
+                if (i < (finger_count - 1))
+                    touchpad_buffer[i] = touchpad_buffer[i + 1];
+            }
+        }
+        if (finger_count > 0)
+            finger_count--;
+        break;
+    case SDL_CONTROLLERTOUCHPADMOTION:
+        for (int i = 0; i < finger_count; i++) {
+            if (touchpad.finger == touchpad_buffer[i].finger) {
+                const auto touch_id = touchpad_buffer[i].which;
+                touchpad_buffer[i] = touchpad;
+                touchpad_buffer[i].which = touch_id;
+            }
+        }
+        break;
+    }
+ 
+    is_touchpad = finger_count;
```

Comment:
`is_touchpad = finger_count > 0;` implicit int -> bool conversion does not look great here.

Reply:
Is see

Reply:
Done